examples/with-session/src/app/page.tsx

```diff
@@ -0,0 +1,144 @@
+'use client'
+import { QRCodeSVG } from 'qrcode.react'
+import { useState, useEffect, useRef } from 'react'
+import { Session, VerificationState, type ISuccessResult, type AppErrorCodes } from '@worldcoin/idkit-core'
+
+type SessionStatus = {
+	state: VerificationState
+	result: ISuccessResult | null
+	errorCode: AppErrorCodes | null
+}
+
+export default function Home() {
+	const sessionRef = useRef<Session | null>(null)
+	const [sessionStatus, setSessionStatus] = useState<SessionStatus | null>(null)
+	const [sessionURI, setSessionURI] = useState<string | null>(null)
+
+	useEffect(() => {
+		// Initialize session only once
+		if (!sessionRef.current) {
+			;(async () => {
+				const session = await new Session().create(
```

Comment:
I feel like this would be nicer if it was more strongly typed. Would make the experience better IMO
```suggestion
				const session = await new Session().create({
				     appId: '',
				     action: '',
				     signal: ''
				}
```